contracts/periphery/astETH.sol

```diff
@@ -0,0 +1,34 @@
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract astETH is ERC20 {
+    using SafeERC20 for IERC20;
+
+    address public immutable stETH;
+    address public immutable feeRecipient;
+
+    constructor(
+        address _stETH,
+        address _feeRecipient
+    ) ERC20("Alongside stETH", "astETH") {
+        stETH = _stETH;
+        feeRecipient = _feeRecipient;
+    }
+
+    function deposit(uint256 amount) external {
+        IERC20(stETH).safeTransferFrom(msg.sender, address(this), amount);
+        _mint(msg.sender, amount);
+    }
```

Comment:
Not a security issue, but if you're making users do this before issuing AMKT, minting stETH, then approving this contract, then calling this is a lot of steps.

It would be a lot smoother if they just sent ETH and this function wrapped it into stETH and minted for them, like this:
```
function deposit() payable external {
    uint balanceBefore = IERC20(stETH).balanceOf(address(this));
    stETH.call{value: msg.value}();
    require(IERC20(stETH).balanceOf(address(this)) - balanceBefore == msg.value);
    _mint(msg.sender, msg.value);
}
```

Reply:
I'm leaning towards not enabling this shortcut purely because it does not increase security, while opening up the small possibility in the future of bricking `deposit` in case the `require` statement fails.